pkg/operator/operator.go

```diff
@@ -0,0 +1,205 @@
+		Namespace:              controller.OperatorNamespace,
+		NewClient:              newNoCacheClientFunc,
+		// Use a non-caching client everywhere. The default split client does not
+		// promise to invalidate the cache during writes (nor does it promise
+		// sequential create/get coherence), and we have code which (probably
+		// incorrectly) assumes a get immediately following a create/update will
+		// return the updated resource. All client consumers will need audited to
+		// ensure they are tolerant of stale data (or we need a cache or client that
+		// makes stronger coherence guarantees).
+		// https://pkg.go.dev/sigs.k8s.io/controller-runtime#hdr-Clients_and_Caches
+	})
+	if err != nil {
+		setupLog.Error(err, "unable to create manager")
+		os.Exit(1)
+	}
+	if err := mgr.AddHealthzCheck("healthz", healthz.Ping); err != nil {
+		setupLog.Error(err, "unable to set up health check")
+		os.Exit(1)
+	}
+	if err := mgr.AddReadyzCheck("readyz", healthz.Ping); err != nil {
+		setupLog.Error(err, "unable to set up ready check")
+		os.Exit(1)
+	}
+
+	if err := (&nodeobservabilitycontroller.NodeObservabilityReconciler{
+		Client:     mgr.GetClient(),
+		Scheme:     GetOperatorScheme(),
+		Log:        ctrl.Log.WithName("controller.nodeobservability"),
+		Namespace:  controller.OperatorNamespace,
+		AgentImage: controller.AgentImage,
+	}).SetupWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to create controller", "controller", "nodeobservability")
+		return nil, fmt.Errorf("failed to create nodeobservability controller: %w", err)
+	}
+	// Cluster is a runnable managed by controller-runtime's manager
+	// with its own client and cache.
+	// We deliberately use a dedicated cache for the watches of ca-configmap-controller
+	// to avoid the unnecessary cluster level permissions.
+	// ca-configmap-controller needs a namespace different from the operator's (source namespace)
+	// which forces the usage of the multinamespace shared cache of the manager.
+	// The additional namespace in turn obliges all the other controllers to have the rights
+	// to watch in it which results into cluster level permissions for all the operands:
+	// daemonset, serviceaccounts, services, since the local role cannot be created in another namespace via OLM.
+	// Inspired by https://github.com/kubernetes-sigs/controller-runtime/blob/master/designs/move-cluster-specific-code-out-of-manager.md
+	cluster, err := cluster.New(config, func(opts *cluster.Options) {
+		opts.NewClient = newNoCacheClientFunc
+	})
+	if err != nil {
+		setupLog.Error(err, "unable to create controller-runtime cluster")
+		os.Exit(1)
+	}
+
+	if err := mgr.Add(cluster); err != nil {
+		setupLog.Error(err, "unable to add controller-runtime cluster to manager")
+		os.Exit(1)
+	}
+
+	if err := machineconfigcontroller.New(mgr).SetupWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to create controller", "controller", "nodeobservabilitymachineconfig")
```

Comment:
Same: no need to duplicate the log message as it'll be in `main.go`. Same is true for the NobRun controller below.